_posts/2015-02-04-9.markdown

```diff
@@ -0,0 +1,191 @@
+---
+layout: post
+title:  "Writing Your First Interaction"
+date:   2015-02-04 10:00:09
+answerrostag: "challenge_rocon_interaction"
+track: [challenge]
+---
+
+[comment]: <> (TODO: replace answerrostag and track)
+
+In this lesson we will write user side of rocon application. We will learn how
+to add follower application in application list.
+
+## Prerequisites
+
+This lesson is a continuation of the previous lesson:
+["Writing Your First Rapp"]({{ site.url }}{% post_url 2015-02-04-8 %}).
+If you have not done this lesson, do it firstly.
+
+## Creating interaction
+
+1\. Create interactions folder in `my_rapps`.
+{% highlight sh %}
+cd ~/helloworld/turtlebot/my_rapps/src/my_rapps/
+mkdir interactions
+{% endhighlight %}
+
+2\. Create `my.interactions` file with following content:
+{% highlight sh %}
+- name: ""
+  role: My Interactions
+  compatibility: rocon:/pc/*/hydro|indigo/precise|quantal|raring|saucy|trusty
+  display_name: Follower
+  description: Take the TurtleBot for a stroll. TurtleBot will follow  whatever is in front of it.
+  max: -1
+  icon:
+    resource_name: my_rapps/follower_bubble_icon.png
+  pairing:
+    rapp: my_rapps/follower
+    remappings: []
+{% endhighlight %}
+
+## Creating my_concert_service_gazebo package
+
+1\. Create package `my_concert_service_gazebo`.
+{% highlight sh %}
+cd ~/helloworld/turtlebot/
+mkdir -p my_concert_service_gazebo/src/my_concert_service_gazebo
+cd my_concert_service_gazebo/src/my_concert_service_gazebo
+{% endhighlight %}
+
+2\. Copy `concert_service_gazebo` folder from
+[github](https://github.com/robotics-in-concert/concert_services/tree/indigo)
+(branch `indigo`).
+
+3\. Edit `robot.launch` file.
+{% highlight sh %}
+cd ~/helloworld/turtlebot/my_concert_service_gazebo/src/my_concert_service_gazebo/launch/
+{% endhighlight %}
+
+Add  `robot_interactions` and `robot_interactions_list` arguments below `robot_concert_whitelist`.
+{% highlight sh %}
+<arg name="robot_interactions" default="false"/>
+<arg name="robot_interactions_list" default="[]"/>
+{% endhighlight %}
+
+Add `interactions` and `interactions_list` arguments below `concert_whitelist`.
+{% highlight sh %}
+<arg name="interactions" value="$(arg robot_interactions)"/>
+<arg name="interactions_list" value="$(arg robot_interactions_list)"/>
+{% endhighlight %}
+
+4\. Edit `gazebo_robot_manager.py` file.
```

Comment:
It would be good to explain why you edit this file in a sentence or two. SInce it's needed to pass the arguments through, otherwise you get the default arguments.